### PR TITLE
Improve memory and cache logging

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Improve memory and cache logging to ease debugging:
use succintBytes for byte and add pages when log memory error
refactor cache logging and use it in Prestissimo side for background logging

### CODE REVIEW

File: velox/common/memory/MallocAllocator.cpp [hunk 1]
```diff
@@ -50,10 +50,11 @@ bool MallocAllocator::allocateNonContiguousWithoutRetry(
     try {
       reservationCB(bytesToAllocate, true);
     } catch (std::exception& e) {
-      VELOX_MEM_LOG(WARNING) << "Failed to reserve " << bytesToAllocate
-                             << " bytes for non-contiguous allocation of "
-                             << numPages << " pages, then release "
-                             << freedBytes << " bytes from the old allocation";
+      VELOX_MEM_LOG(WARNING)
+          << "Failed to reserve " << succinctBytes(bytesToAllocate)
+          << " bytes for non-contiguous allocation of " << numPages
```
Comment (tanjialiang) [inline velox/common/memory/MallocAllocator.cpp:55]: remove "bytes"

File: velox/common/memory/MallocAllocator.cpp [hunk 2]
```diff
@@ -304,7 +307,7 @@ void* MallocAllocator::allocateBytesWithoutRetry(
   void* result = (alignment > kMinAlignment) ? ::aligned_alloc(alignment, bytes)
                                              : ::malloc(bytes);
   if (FOLLY_UNLIKELY(result == nullptr)) {
-    VELOX_MEM_LOG(ERROR) << "Failed to allocateBytes " << bytes
+    VELOX_MEM_LOG(ERROR) << "Failed to allocateBytes " << succinctBytes(bytes)
                          << " bytes with " << alignment << " alignment";
```
Comment (tanjialiang) [inline velox/common/memory/MallocAllocator.cpp:311]: remove "bytes"

**Review by tanjialiang — COMMENTED** [review]
Thanks for the improvement. Some minors

**Review by tanjialiang — APPROVED** [review]

File: velox/common/caching/AsyncDataCache.cpp [hunk 3]
```diff
@@ -654,7 +654,7 @@ void AsyncDataCache::backoff(int32_t counter) {
   size_t seed = folly::hasher<uint16_t>()(++backoffCounter_);
   const auto usec = (seed & 0xfff) * (counter & 0x1f);
   VELOX_CACHE_LOG_EVERY_MS(INFO, 1'000)
-      << "Backoff in allocation contention for " << usec << " us.";
+      << "Backoff in allocation contention for " << succinctMicros(usec);
```
Comment (mbasmanova) [inline velox/common/caching/AsyncDataCache.cpp:657]: I wonder what does "Backoff in allocation contention" mean? Would you explain a bit?

**Review by mbasmanova — COMMENTED** [review]
Thanks.

**Pull request closed without merging**